### ISSUE TITLE
Playing a folder of songs / creating a playlist

### ISSUE THREAD

**Issue by pjrobertson** [pdherbemont/Glasses#62]
Dragging a folder containing MP3s etc. onto the lunettes icon doesn't create a playlist of all the songs in the folder like VLC does (I can't see a way of viewing all the songs in lunettes, it only seems to show the currently playing)

Also, after adding a folder, it'll only play the first song in that folder. Once the 1st song has finished, if you click 'play' the second will start playing but Lunettes will then crash (and become unresponsive until you force quit).
Here's the error from the console:

02/04/2010 09:41:53 Lunettes[9103] NSExceptionHandler has recorded the following exception: NSRangeException -- Cannot remove an observer <NSKeyValueObservance 0x118ba2a00> for the key path "media.@count" from <VLCMediaList 0x118ab92f0> because it is not registered as an observer. Stack trace: 0x7fff85662a2c 0x7fff807f70f3 0x7fff84d61b47 0x7fff84d61ad4 0x7fff85cebef2 0x7fff85cebc3f 0x7fff85cec522 0x7fff85cebd5b 0x7fff85cebc3f 0x7fff81f09107 0x7fff869beb0c 0x7fff869bfdbf 0x7fff869b1a68 0x10002656a 0x10002760a 0x1000170e1 0x100010b45 0x10000fb09 0x7fff86bfda67 0x1000051f9 0x10000e1a2 0x7fff86b208ea 0x7fff86b44e5a 0x7fff86b44bc4 0x7fff86b2a859 0x7fff86b29601 0x7fff869fa221 0x10001f743 0x7fff869910aa 0x7fff86989d7c 0x100001e75 0x100001c14 0x2 02/04/2010 09:41:53 Lunettes[9103] NSExceptionHandler will now suspend the current thread and wait for debugger attachment

P.S. the wiki says to report issues to the VL trac, but nobody seems to be looking at those there

**Comment by SnidelyW** [comment]
Hello PJ.

Hmmm...I just dragged a folder of mp3's onto the Lunettes app icon. Lunettes launched, and opened a SECOND window on top of the regular spashwindow where video files are listed. This second window has the name of the folder containing the mp3's and is divided into two sections. The left portion is a black window where video would normally play. The right portion lists all of the songs that are in the mp3 folder that I dragged on top of the Lunettes icon. I can play any song in the list simply by double-clicking it. After the first song in the list finished, I clicked on the forward button that is next to the play button. The second song in the playlist immediately began to play without a problem. Lunettes has not crashed.

Are you using the very latest build of Lunettes, i.e., test4?

**Comment by pjrobertson** [comment]
aah test4 seems to be working.

All looks good thanks :)

The only thing now is that it doesn't continue playing to the next track at the end of the 1st, and the net track / previous track buttons (F9, F7) don't seem to work

**Comment by SnidelyW** [comment]
Great! Glad to hear that your problem was resolved. Regarding non-continuous play, I agree; that is something that needs to be fixed/added.

Regarding the function key issue, I don't know if you are aware of it or not, but for a number of months now, there has been quite a howl on the Apple.com site, because folks are not happy with the fact that the media keys are hard-coded/mapped to iTunes, because Apple only wants everyone to use iTunes, and not 3rd party media players like VLC, Lunettes, Mplayer, Chroma Player, etc.

While I could be wrong, I suspect that this may not be an issue that Pierre and Felix can fix as long as Apple is monopolizing those keys. There are a few workarounds on the Apple.com site, but they are a bit nasty and involve Applescripts and such, or moving files out of their default locations, etc.

**Comment by pjrobertson** [comment]
Yeah I know about that business - pressing the play button makes iTunes launch as well as Lunettes to pause etc.

My problem is that pressing the next/back keys once doesn't do anything. If I hold them it'll scroll through the track, but pressing once doesn't go to the next track (like it does in iTunes, VLC etc.)

Also the problem of not playing the next track having finished one (continuing on)

**Comment by pjrobertson** [comment]
The problem seems to be that there actually aren't any next/back buttons at all:

None in the menu bar, none in the GUI (If I select the TV style there's only play and pause) and the keyboard buttons don't work.

P.S. just seen a commit on 'better drag and drop'
Merci Pierre ;)

**Comment by SnidelyW** [comment]
Ha! I was just going to say that I hope that Pierre and Felix visit this Issues page soon and take a peek at what is going on here. Going by your above remarks, maybe they already have. I just got two new emails from Pierre. I am going to see what he has to say now.  :)

**Comment by pdherbemont** [comment]
Disclaimer: Making Lunettes a premium player for music is not a priority for me.

**Comment by pdherbemont** [comment]
Works here. I have some stuff to push though.

**Comment by pjrobertson** [comment]
Thanks,

The latest commit seems to support drag & drop OK now but a playlist still doesn't appear unless you change to 'default' style

With VLCs good support for various formats, it's one of the only apps that'll run FLACs on Macs so without good music / FLAC support it'd be hopeless!

Thanks anyway

**Comment by SnidelyW** [comment]
PJ, while I have been an avid Mac user since 1990, it might interest you to know that I very rarely ever use QuickTime for watching video -- sometimes I use it to export -- or iTunes for listening to music.

Some time ago I discovered a very nice, free music player which now goes by the name of Vox. It is small, light-weight and probably has most of the features that you need in a basic music player -- and yes, it plays FLAC files just fine.

You can access the developer's blog and download page at the following URL:

http://alenofx.viviti.com/

There is also a support forum at this URL:

http://voxapp.didgeroo.com/forums/

BTW, the developer is very attentive to user feedback.
